src/types.ts

```diff
@@ -5,27 +5,24 @@ import {
   Plugin,
   AppConfig,
   VNode,
-  VNodeProps
+  VNodeProps,
+  FunctionalComponent
 } from 'vue'
 
-interface RefSelector {
+export interface RefSelector {
   ref: string
 }
-
-interface NameSelector {
-  name: string
-}
-
-interface RefSelector {
-  ref: string
-}
-
-interface NameSelector {
+export interface NameSelector {
   name: string
+  length?: never
```

Comment:
This was a nasty one. `{ name: string }` definition matches Function (because functions have `name` property) which messes with our type inference. Putting `length` to `never` allows us to distinguish between finding a function and finding by object with name field

Reply:
damn, nasty but smart. thanks for the explanation, I would've missed it